### PR TITLE
Using render HOC to initialize session

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Utilize HOC to initialize the session instead of calling a mutation

#### What problem is this solving?
Using the new HOC provided by the render

#### How should this be manually tested?
[workspace](https://felipe1--storecomponents.myvtex.com)

### Linked to:
[#PR108](https://github.com/vtex-apps/store-graphql/pull/108)
[#PR126](https://github.com/vtex-apps/render-runtime/pull/126)

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/index.js [hunk 1]
```diff
@@ -31,48 +34,55 @@ class Telemarketing extends Component {
...
   }
 
-  componentDidMount = () => {
-    requestWithRetry(this.props.initializeSession)
-      .then(res => this.processSession(res.data && res.data.initializeSession))
-      .catch(err => console.log('err', err))
-  }
+  static getDerivedStateFromProps(props, state) {
+    const shouldProcessSession =
+      state.isLoadingSession === true && props.session.loading === false
```
Comment (lucasecdb) [inline react/index.js]: You can use `state.isLoadingSession && !props.session.loading`

File: react/queries/getSession.gql [hunk 2]
```diff
@@ -9,10 +9,8 @@ mutation initSession {
       storeUserEmail
```
Comment (brunojdo) [inline react/queries/getSession.gql]:
```
impersonate { 
   profile {...} 
}
```
I also removed `storeUserId` and `storeUserEmail`

File: react/utils/processSession.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+export default session => {
+  const {
+    adminUserEmail,
+    impersonate: { storeUserId, storeUserEmail },
+    impersonable,
+    profile: { document, firstName, lastName, phone },
+  } = session
+
+  return {
+    logged: storeUserId ? true : false,
```
Comment (brunojdo) [inline react/utils/processSession.js]: check if the profile exists in impersonate.

File: react/utils/processSession.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    canImpersonate: impersonable,
+    clientDocument: document || '',
```
Comment (brunojdo) [inline react/utils/processSession.js]: Don't think this map is neccessary

File: react/index.js (hunk 1 above)
Comment (salesfelipe) [reply react/index.js]: 👍

**Pull request merged**